tests/test_outputs/test_form_output_prep.py

```
@@ -0,0 +1,329 @@
+        """
+        columns = [
+            "Unnamed: 0",
+            "SIC 2007_CODE",
+            "2016 > Form PG"
+        ]
+        data = [
+            [100, 10130, 3],
+            [198, 14190, 6],
+            [223, 17219, 9],
+            [253, 20120, 12],
+            [311, 23610, 15],
+            [772, 46390, 40],
+            [774, 46420, 40],
+            [1088, 71129, 50],
+       ]
+        sic_pg_num = pd.DataFrame(data=data, columns=columns)
+        return sic_pg_num
+    
+
+    @pytest.fixture(scope="function")
+    def pg_num_alpha(self) -> pd.DataFrame:
+        """'pg_num_alpha' input for form_output_prep.
+        
+        NOTE: This is a subset of columns for testing purposes
+        """
+        columns = [
+            "Unnamed: 0",
+            "pg_numeric",
+            "pg_alpha"
+        ]
+        data = [
+            [2, 3, 'C'],
+            [5, 6, 'D'],
+            [8, 9, 'E'],
+            [11, 12, 'G'],
+            [14, 15, 'J'],
+            [39, 40, 'AA'],
+            [48, 50, 'AD'],
+       ]
+        pg_num_alpha = pd.DataFrame(data=data, columns=columns)
+        return pg_num_alpha
+    
+
+    @pytest.fixture(scope="function")
+    def ni_expected(self) -> pd.DataFrame:
+        columns = [
+            'reference',
+            'instance',
+            'ultfoc',
+            '407',
+            '201',
+            'rusic',
+            'a_weight',
+            '604',
+            'form_status',
+            '602',
+            'formtype',
+            'pg_numeric'
+        ]
+        data = [
+            [39900001577, 1, 'GB ', 0.72, 'C', 10130, 1, 'Yes', 600, 100, "0003", 3.0],
+            [39900006060, 1, 'GB ', 0.6, 'G', 20120, 1, 'Yes', 600, 100, "0003", 12.0],
+            [39900008752, 1, 'GB ', 0.72, 'AA', 46420, 1, 'Yes', 600, 100, "0003", 40.0],
+            [39900008762, 1, 'GB ', 6.9, np.nan, 22230, 1, 'Yes', 600, 100, "0003", np.nan],
```

we shouldn't need an example with null in the pg columns 201 and pg_numeric as when we join to a mapper we should throw an error if there is no match.

---

removed